src/components/Query.ts

```diff
@@ -174,12 +174,19 @@ export class APIRequest {
         query : string
     ) : Promise<QueryResult> {
         let data : string = ''
+        let url : string = `${conn.hostNport}/api/v2/query`
         try {
             this.source = axios.CancelToken.source()
+            if (conn.version == InfluxConnectionVersion.V2) {
+                url = `${url}?org=${encodeURI(conn.org)}`
+            } else if (conn.version == InfluxConnectionVersion.V1) {

```

Comment:
Since this method is called `queryV2` should there be a separate method for `V1`?

Reply:
I thought about that, but I think the name refers to the `v2` endpoint that both 1.x and 2.0 have at `api/v2/`, so this function works for both. Maybe it should just be called `query`?

Reply:
And yes, the username and password are only used in 1.x connections. 2.0 connections use org and token for auth.

Reply:
I am very surprised that 1.x has an endpoint at `api/v2` but if that's the case, what you have here is fine. I assume that you verified this against OSS 1.x?

Reply:
I did!

Reply:
For reference: https://docs.influxdata.com/influxdb/v1.8/tools/api/

Reply:
Having a second look at those docs... maybe it's a better idea to include the username and password in the `Token` header instead of the url. Thoughts?

Reply:
I think the header is a better place for it. My understanding is that the query string is part of the URL and will not be encrypted if you are using https---the docs also seem to suggest this is the right way.

Reply:
Agreed. I'll change that now.